mods/fire/init.lua

```diff
@@ -292,9 +292,18 @@ minetest.register_abm({
 
 local fire_enabled = minetest.setting_getbool("enable_fire")
 if fire_enabled == nil then
-	-- New setting not specified, check for old setting.
-	-- If old setting is also not specified, 'not nil' is true.
-	fire_enabled = not minetest.setting_getbool("disable_fire")
+	-- New setting not specified, check for old setting
+	fire_enabled = minetest.setting_getbool("disable_fire")
+	if fire_enabled == nil then
+		-- Neither setting specified, check whether singleplayer
+		if minetest.is_singleplayer() then
+			fire_enabled = true
+		else
+			fire_enabled = false
+		end
+	else
+		fire_enabled = not fire_enabled
```

Comment:
This is a confusing way to do this since `fire_enabled` actually means "fire disabled" until after this line.  I'd do something like this:
```Lua
local disabled = minetest.settings:get_bool("disable_fire")
if disabled == nil then
        fire_enabled = minetest.is_singleplayer()
else
        fire_enabled = not disabled
end
```

You could also flatten it out a bit if you're not above doing two settings reads:
```Lua
-- Use enable_fire, but fall back to disable_fire for compatibility.
-- Enable only in single player mode if neither is set.
local fire_enabled = minetest.settings:get_bool("enable_fire")
local disabled_compat = minetest.settings:get_bool("disable_fire")
if fire_enabled == nil and disabled_compat == nil then
        fire_enabled = minetest.is_singleplayer()
else if fire_enabled == nil then
        fire_enabled = not disabled_compat
end
```